### PR TITLE
Puts contributing, issue template, support, and CoC in subdir

### LINKED ISSUE #165: Put GitHub md files in .github/
Should we?

https://github.com/blog/2111-issue-and-pull-request-templates

https://help.github.com/articles/setting-guidelines-for-repository-contributors/

https://help.github.com/articles/creating-an-issue-template-for-your-repository/

### PR DESCRIPTION
**Pull request by batpigandme**
* Fixes #165
* Puts contributing guidelines, issue template, support, and code of conduct in `.github` subdirectory
* Moves ignore parameter to .github directory rather than individual files
* Tests for presence of .github directory in tests
* Adds use_tidy_coc() for code of conduct in .github directory (basically an alias for use_code_of_conduct())
* Updates documentation for same
* Adds bullet to NEWS

### CODE REVIEW

File: R/tidyverse.R [hunk 1]
```diff
@@ -161,9 +167,38 @@ use_tidy_issue_template <- function() {
...
+
+  use_directory(".github", ignore = TRUE)
+  use_template(
+    "CODE_OF_CONDUCT.md",
+    ".github/CODE_OF_CONDUCT.md"
+  )
+
+  todo("Don't forget to describe the code of conduct in your README.md:")
+  code_block(
+    "Please note that this project is released with a [Contributor Code of Conduct](CODE_OF_CONDUCT.md).",
```
Comment (jennybc) [inline R/tidyverse.R]: Does the relative link now need to be `.github/CODE_OF_CONDUCT.md`?

File: R/tidyverse.R [hunk 2]
```diff
@@ -26,6 +26,12 @@
 #' * `use_tidy_support()`: creates support resources document for GitHub repo
 #'    using SUPPORT.md.
 #'
+#' * `use_tidy_coc()`: creates code of conduct same as use_code_of_conduct
+#'    but for the fact that it's placed in a .github subdirectory.
+#'
+#' *  `use_tidy_community`: wrapper function around contributing, issues,
+#'     support, and code of conduct functions.
+#'
```
Comment (jennybc) [inline R/tidyverse.R]: Can you consistently add `()` when referring to function and enclose in backticks?

**Comment by jennybc** [comment]
While you're kicking around in this file, it sure seems like `use_tidy_contributing()` should have `check_uses_github()` for internal consistency, yes? If you agree, please add that.

File: R/tidyverse.R (hunk 2 above)
Comment (batpigandme) [reply R/tidyverse.R]: (>ლ)
Yes!
Sorry, Jenny, you've literally told me that exact same thing before…

File: R/tidyverse.R [hunk 3]
```diff
@@ -19,18 +19,18 @@
...
-#'    tidy_contributing template.
+#'   tidy_contributing template.
 #'
 #' * `use_tidy_issue_template()`: creates a standard tidyverse issue template.
 #'
 #' * `use_tidy_support()`: creates support resources document for GitHub repo
-#'    using SUPPORT.md.
+#'   using SUPPORT.md.
 #'
 #' * `use_tidy_coc()`: creates code of conduct same as use_code_of_conduct
```
Comment (jennybc) [inline R/tidyverse.R]: "use_code_of_conduct" still needs the backtick-and-parens treatment 😅

I'm not giving up this time!

Comment (batpigandme) [reply R/tidyverse.R]: _OH_! Ok, ok! I see what you mean. I actually _did_ have that at one point, which I think just arbitrarily coincided with my getting an error when I ran R CMD Check, and I just assumed they were related!

**Comment by hadley** [comment]
Do we plan to make an umbrella `use_tidy_github()`?

**Comment by batpigandme** [comment]
Do you mean in addition to `use_tidy_community()`? (i.e. beyond just setting up the `.github` subdirectory with the four files)

**Comment by hadley** [comment]
Oops I missed that - I was thinking that would be a replacement of, rather than a supplement to, the individual functions. (i.e. we wouldn't expect `use_tidy_coc()` etc)

**Comment by batpigandme** [comment]
Ah, OK. I'm happy to do it that way, my thinking was basically that it's _possible_ that someone would want to use, say, the code of conduct, but not the issue template (e.g. I presume ggplot2 will continue to keep its own package-specific template, at least in part). I suppose since these don't overwrite existing files, though, that could be done even with a unified function.

**Comment by hadley** [comment]
I think in the future, we'll run them earlier and then customise, and for current packages you can always override and then roll back with git

**Comment by batpigandme** [comment]
FTR, I did this (in a new branch), and it _should_ work, but I can no longer get usethis to check without a note re. `Missing or unexported object: ‘git2r::head’`. (>ლ)

**Pull request merged**